framework/src/SPIDevObj.cpp

```diff
@@ -154,7 +193,36 @@ int SPIDevObj::writeRegVerified(DevHandle &h, uint8_t address, uint8_t val)
 
 int SPIDevObj::_writeReg(uint8_t address, uint8_t val)
 {
-	uint8_t write_buffer[2];
+#ifdef __RPI2
+    /* implement sensor interface via rpi2 spi */
+    // constexpr int transfer_bytes = 1 + 1; // first byte is address
+    uint8_t write_buffer[2] = {0}; // automatic write buffer
+    uint8_t read_buffer[2] = {0}; // automatic read buffer
+
+    struct spi_ioc_transfer spi_transfer; // datastructures for linux spi interface
+    memset(&spi_transfer, 0, sizeof(spi_ioc_transfer));
+
+    write_buffer[0] = address | DIR_WRITE; // write mode
+    write_buffer[1] = val; // write data
+
+    spi_transfer.rx_buf = (unsigned long)read_buffer;
+    spi_transfer.len = 2;
+    spi_transfer.tx_buf = (unsigned long)write_buffer;
+    // spi_transfer.speed_hz = SPI_FREQUENCY_1MHZ; // temporarily override spi speed
+    spi_transfer.bits_per_word = 8;
+    spi_transfer.delay_usecs = 0;
+
+    int result = 0;
+    result = ::ioctl(m_fd, SPI_IOC_MESSAGE(1), &spi_transfer);
+
+    if (result != 2) {
+        DF_LOG_ERR("Error: SPI write failed. Reported %d bytes written (%d)", result, errno);
+        return -1;
+    }
+    return 0;
+#else
+
+    uint8_t write_buffer[2];
```

Comment:
@julianoes can you point to me which formatter i need to run? I used to run astyle tool of the Firmware (`fix_code_style.sh`, default setting parameters in Firmware). but it generated worse indentations.